apps/dashboard/app/views/layouts/application.html.erb

```diff
@@ -37,10 +37,22 @@
 
       <div class="collapse navbar-collapse" id="navbar">
         <ul class="navbar-nav mr-auto">
-          <%= render partial: 'layouts/nav/featured_apps', locals: { group: @featured_group } if @featured_group.present? %>
-          <%= render partial: 'layouts/nav/group', collection: @nav_groups %>
-          <%= render partial: 'layouts/nav/sessions', nav_groups: @nav_groups if Configuration.app_development_enabled? || @nav_groups.any?(&:has_batch_connect_apps?) %>
-          <%= render partial: 'layouts/nav/all_apps' if @user_configuration.show_all_apps_link %>
+          <% if @nav_bar.empty? %>
+            <%= render partial: 'layouts/nav/featured_apps', locals: { group: @featured_group } if @featured_group.present? %>
+            <%= render partial: 'layouts/nav/group', collection: @nav_groups %>
+            <%= render partial: 'layouts/nav/sessions', nav_groups: @nav_groups if Configuration.app_development_enabled? || @nav_groups.any?(&:has_batch_connect_apps?) %>
+            <%= render partial: 'layouts/nav/all_apps' if @user_configuration.show_all_apps_link %>
+          <% else %>
+            <% @nav_bar.each do |nav_item| %>
+              <%= begin
+                    render partial: nav_item.partial_path, locals: nav_item.to_h
+                  rescue NoMethodError => e
+                    raise StandardError, nav_item.inspect
+                    # raise e
+                  end
```

Comment:
I had this for testing, I'm not sure if we still need the rescue (was catching some delegator issues I was having).